wagtail_content_import/mappers/converters.py

```diff
@@ -47,10 +51,54 @@ def contentstate_converter(self):
             features = self.features
         return ContentstateConverter(features=features)
 
+    @cached_property
+    def site_root_paths(self):
+        return Site.get_site_root_paths()
+
+    def convert_external_links(self, html):
+        rewriter = LinkRewriter({'external': self.convert_external_link_tag})
+        return rewriter(html)
+
+    def convert_external_link_tag(self, attrs):
+        # Convert any external link tags that exactly match internal urls to page links
+        href = attrs.get('href', '')
+        page = self.get_page_for_url(href)
+        if page:
+            return f'<a linktype="page" id="{page.pk}">'
+        return f'<a linktype="external" href="{href}">'
+
+    def get_page_for_url(self, submitted_url):
+        # Strip the url of its query/fragment link parameters - these won't match a page
+        url_without_query = re.split(r"\?|#", submitted_url)[0]
+
+        if url_without_query != submitted_url:
+            # We only want to convert exact matches
+            return
+
+        # Start by finding any sites the url could potentially match
+        sites = self.site_root_paths
+
+        possible_sites = [
+            (path, url_without_query[len(url) + 1:])
+            for pk, path, url, language_code in sites
+            if submitted_url.startswith(url)
+        ]
+
+        # Loop over possible sites to identify a page match
+        for root_path, url in possible_sites:
+            matched_pages = Page.objects.filter(url_path=root_path + url).specific()
```

Comment:
Do we need ``.specifc()`` here? (looks like we just need its ``.pk``?)

Reply:
You're right - we had `specific` before because I was calling page methods they could have customised to double check its url matched exactly, but since I now do that check elsewhere, it's no longer necessary